core/src/main/scala/com/nvidia/spark/rapids/tool/Platform.scala

```diff
@@ -148,7 +151,7 @@ abstract class Platform(var gpuDevice: Option[GpuDevice]) {
 
 abstract class DatabricksPlatform(gpuDevice: Option[GpuDevice]) extends Platform(gpuDevice) {
   override val defaultGpuDevice: GpuDevice = T4Gpu
-
+  override val defaultGpuForSpeedupFactor: GpuDevice = T4Gpu
```

Comment:
is there a reason to add these explicitly? The default is to just use the defaultGpuDevice so this seems like its unneeded if they are the same.

Reply:
The reason I added this is for cases like `databrick-aws-l4`, where the tool cannot find operator score file for this platform + gpu combination, so it tries to use the default gpu for getting scores, the most reasonable variable is `defaultGpuForSpeedupFactor`, which is defined as:
```
def defaultGpuForSpeedupFactor: GpuDevice = getGpuOrDefault
final def getGpuOrDefault: GpuDevice = gpuDevice.getOrElse(defaultGpuDevice)
```
However, if the platform does not have `defaultGpuForSpeedupFactor` defined, it falls back to `gpuDevice` before falling back to`defaultGpuDevice`, which is the incorrect `l4` in the case.
Another way to approach is to change the definition of `defaultGpuForSpeedupFactor` to `defaultGpuDevice`.

Reply:
I thought more about this and I updated `defaultGpuForSpeedupFactor` definition to:
```
def defaultGpuForSpeedupFactor: GpuDevice = defaultGpuDevice
```
and removed the redundant assignments to `defaultGpuForSpeedupFactor`.